### ISSUE TITLE
'halID' referenced before assignment when using --restart (during cactus-align)

### ISSUE THREAD

**Issue by brantfaircloth** [ComparativeGenomicsToolkit/cactus#368]
Good afternoon,

I've run into a problem that seems to occur when restarting `cactus-align` while running using the stepwise approach.  The gist of the problem is that the workflow seems to run fine after `--restart` until it gets to the point of writing the hal file.  The workflow seems to complete, but hits the error:

```
qbc030 2020-12-05 11:25:03,773 MainThread INFO toil.leader: Finished toil run successfully.
Traceback (most recent call last):
  File "/opt/python/miniconda3/bin/cactus-align", line 8, in <module>
    sys.exit(main())
  File "/opt/python/miniconda3/lib/python3.7/site-packages/cactus/setup/cactus_align.py", line 122, in main
    runCactusAfterBlastOnly(options)
  File "/opt/python/miniconda3/lib/python3.7/site-packages/cactus/setup/cactus_align.py", line 259, in runCactusAfterBlastOnly
    toil.exportFile(halID, makeURL(options.outputHal))
UnboundLocalError: local variable 'halID' referenced before assignment
```
Running one more time with `--restart` outputs:
```
qbc001 2020-12-05 12:20:08,306 MainThread WARNING toil.common: Requested restart but the workflow has already been completed; allowing exports to rerun.
Traceback (most recent call last):
  File "/opt/python/miniconda3/bin/cactus-align", line 8, in <module>
    sys.exit(main())
  File "/opt/python/miniconda3/lib/python3.7/site-packages/cactus/setup/cactus_align.py", line 122, in main
    runCactusAfterBlastOnly(options)
  File "/opt/python/miniconda3/lib/python3.7/site-packages/cactus/setup/cactus_align.py", line 259, in runCactusAfterBlastOnly
    toil.exportFile(halID, makeURL(options.outputHal))
UnboundLocalError: local variable 'halID' referenced before assignment
```
This seems to *only* occur after using `--restart`. I have a number of stepwise blast/align/hal2fasta jobs that have completed just fine (and entire alignments that have worked fine before) but none of those jobs required a `--restart`. I cannot find any errors in the `stderr` file for this cactus-align job. I took a look at the source near the error, and it does not seem like `halID` is getting set by `toil.restart()` in https://github.com/ComparativeGenomicsToolkit/cactus/blob/b85a75f3daaadedcdecf176e138cea63582afc4b/src/cactus/setup/cactus_align.py#L131-L132
Although I do not know why.

It would be great to be able to move past this issue, generally, but even more excellent if I could export the hal from the current run (it's a rather large job).  I'm on commit 0b6020ff in a Singularity image.

Thank you in advance and happy to provide more details if needed,
-b

**Comment by glennhickey** [comment]
Which version are you using?  As far as I can tell, that bug was fixed in version 1.2.2.

**Comment by brantfaircloth** [comment]
Hi Glenn,

Yup... I am on ~1.0.x.  Sorry for the bother - I just figured out the fix and was able to export the hal after editing the source to:
```
131         if options.restart:
132             halID = toil.restart()
```
Thank you very much for the fast response,
-b
